index.js

```diff
@@ -71,26 +68,18 @@ class Storage {
           return cb(err)
         }
 
-        const isLastChunk = index === this.lastChunkIndex
-        const bytes = new Uint8Array(isLastChunk ? this.lastChunkLength : this.chunkLength)
+        response.arrayBuffer().then((arrayBuffer) => {
+          const buf = Buffer.from(arrayBuffer)
+          if (!opts) return cb(null, buf)
 
-        const reader = response.body.getReader()
+          const offset = opts.offset || 0
+          const len = opts.length || (buf.length - offset)
 
-        let offset = 0
-        reader.read().then(function readChunk ({ done, value }) {
-          if (done) {
-            const buf = Buffer.from(bytes)
-            if (!opts) return cb(null, buf)
-            const offset = opts.offset || 0
-            const len = opts.length || buf.length - offset
-            return cb(null, buf.slice(offset, len + offset))
+          if (opts.offset === 0 && len === buf.length - offset) {
```

Comment:
Can we do this instead:
```suggestion
          if (offset === 0 && len === buf.length) {
```

Reply:
Yup! That's better :)